web/tests/filter.test.js

```diff
@@ -179,8 +181,16 @@ test("basics", () => {
     assert.ok(!filter.is_personal_filter());
     assert.ok(filter.can_bucket_by("channel"));
     assert.ok(filter.can_bucket_by("channel", "topic"));
+    assert.ok(filter.can_bucket_by("channel", "topic", "with"));
     assert.ok(!filter.is_conversation_view());
 
+    terms = [
+        {operator: "channel", operand: "foo"},
+        {operator: "with", operand: 17},
+    ];
+    filter = new Filter(terms);
+    assert.ok(filter.can_bucket_by("channel", "with"));
```

Comment:
Is it correct to say we can bucket by those operators? We don't actually know which messages are in the current conversation... We might need to be careful here.

Reply:
Can you comment on how you addressed this feedback detail? Generally it's best practice for every thread to include explanation of how the feedback was resolved.

Reply:
So in the current version of PR, I only use `can_bucket_by("channel", "topic", "with")` in the PR.

Since the `with` operator is not available in the search suggestion, and is only through either topic-mentions or left_sidebar, I guess it is certain that operations involving `with` only contain the above operands.